fabric-convention-tags-v2/src/main/java/net/fabricmc/fabric/api/tag/convention/v2/ConventionalItemTags.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.tag.convention.v2;
+
+import net.minecraft.item.Item;
+import net.minecraft.registry.tag.TagKey;
+
+import net.fabricmc.fabric.impl.tag.convention.v2.TagRegistration;
+
+/**
+ * See {@link net.minecraft.registry.tag.ItemTags} for vanilla tags.
+ * Note that addition to some vanilla tags implies having certain functionality.
+ */
+public final class ConventionalItemTags {
+	private ConventionalItemTags() {
+	}
+
+	// Natural stone-like blocks that can spawn in-world such as Stone, Granite, Diorite, and others.
+	// Not variations like Stone Bricks or Diorite Slabs.
+	public static final TagKey<Item> STONES = register("stones");
+
+	// Tool tags
+	public static final TagKey<Item> TOOLS = register("tools");
+	public static final TagKey<Item> SHEARS_TOOLS = register("tools/shears");
+	/**
+	 * For throwable spear weapons, like Minecraft's tridents.
+	 * Note, other throwable weapons like boomerangs and throwing knives are best put into their own tools tag.
+	 */
+	public static final TagKey<Item> SPEARS_TOOLS = register("tools/spears");
+	public static final TagKey<Item> BOWS_TOOLS = register("tools/bows");
+	public static final TagKey<Item> CROSSBOWS_TOOLS = register("tools/crossbows");
+	public static final TagKey<Item> SHIELDS_TOOLS = register("tools/shields");
+	public static final TagKey<Item> FISHING_RODS_TOOLS = register("tools/fishing_rods");
```

Comment:
I would be interested in 3 additional grouping tags: `MINING_TOOLS`, `WEAPON_TOOLS` (maybe one for ranged weapons as well?), and `UTILITY/OTHER_TOOLS` to further organize what ultimately ends up in `TOOLS`

Reply:
Hmm. I do want to see what other people's thoughts are on this. This would be basically similar to how Forge/Neo has ToolActions in code. 

Lurkers, comment your thoughts here!

Reply:
As long as they include the tags for the constituent tool types, I wouldn’t mind them that much.

Reply:
Would prefer to expand on this slightly:
- `"tools/weapons"` (includes `"tools/ranged_weapons"` and `"tools/melee_weapons"`)
- `"tools/ranged_weapons"`
- `"tools/melee_weapons"`
- `"tools/mining_tools"`
All these should be included in `"tools"` of course.

Don't see a need for a dedicated utility tools tag, those can simply be included in `"tools"` without further distinction.

Reply:
Gonna poke at this a bit more with some questions. Are these proposed tags for classifying the actual actions the tools do? If so, Neo won't accept it as they are pushing for their modders to use the ToolActions API ([ToolAction](https://github.com/neoforged/NeoForge/blob/55133ebdb40c8e83d8d816b5bdf06c0683c48116/src/main/java/net/minecraftforge/common/ToolAction.java#L14) and [ToolActions](https://github.com/neoforged/NeoForge/blob/55133ebdb40c8e83d8d816b5bdf06c0683c48116/src/main/java/net/minecraftforge/common/ToolActions.java#L16))

So tags that defines what actions a tool actually does probably would go under `fabric` namespace as it would be largely unique to Fabric. It's also probably a symptom of a larger issue on Fabric of a lack of ToolActions API for it.

Currently the tool tags are more for classifying the kind of tool, not the behavior of the tool. Example, someone making a throwable sword should tag it under the swords tag and not the spears tag as it is a sword. Having a sword-looking item under spears tag would be confusing for consumers of that tag.

Thus behavior based tags would probably be a better discussion to have separately on if tag under `fabric` namespace is the way to go or if providing an API way of defining behavior for it would be better.

Reply:
These tags do not define behavior in the sense that adding them to the tag implements them as that tool, like the old (`fabric:pickaxes`) fabric tags did (ie a pickaxe need not be in the tag), but are meant to convey behavior (eg this tool is used like a pickaxe).

AutoSwitch relies on them quite heavily, and exposes them to users for this purpose.

Reply:
@dexman545 What about tools that depend on NBT for what actions they do? Several people in Neoforge said their tools are impossible to properly represent by tags based on tool's actions. I would imagine Fabric also has the same issue but you haven't gotten reports of it yet.

Think like a modular multitool mod where you can swap parts of the tool to allow different actions. It is the same base item but depending on the head attachements, you could have a axe/shear combo or sword/pickaxe combo, etc that's based on NBT.

I can go ahead and add the tool action tags to `c:` for Fabric only but are you sure that an in-code API for what actions a tool can do wouldn't be better and more comprehensive than having it tag based?

Reply:
Tags are sufficient for the majority of items.

autoswitch does have a (limited and somewhat outdated since so few use it) API to handle these cases and a few others.

Reply:
How's this?
https://github.com/FabricMC/fabric/pull/3310/commits/d048b46434b50f1ab414954e7009aa24dca30308

Trident is tricky as it kind of both a ranged and melee weapon but I stuck it into range tag. That alright?

Reply:
Gonna mark this conversation as resolved. If you have any suggestions to adjust these new tool action based tags, just reply and I'll reopen this conversation